ipld/unixfs/mod/dagmodifier_test.go

```diff
@@ -867,3 +872,655 @@ func getOffset(reader uio.DagReader) int64 {
 	}
 	return offset
 }
+
+// makeTestDataPattern generates test data with a repeating pattern
+func makeTestDataPattern(size int, pattern string) []byte {
+	data := make([]byte, size)
+	patternBytes := []byte(pattern)
+	for i := range data {
+		data[i] = patternBytes[i%len(patternBytes)]
+	}
+	return data
+}
```

Comment:
I only see one pattern being used, so might as well not bother passing the pattern. This makes it able to replace more test code.
```suggestion
// makeTestData generates test data with repeating lowercase ascii alphabet.
func makeTestData(size int) []byte {
	data := make([]byte, size)
	for i := range data {
		data[i] = byte('a' + i%26)
	}
	return data
}
```